File: Python_Solutions/reverse_string/str_reverse.py
# Given a string, return a new string with
# reversed order of characters


def str_reverse(string):
    rev = ''
    for ch in range(len(string)):
        rev = string[ch] + rev

    return rev


print(str_reverse('12345'))

# Alternate solution 1
# def str_reverse(string):
#     rev = ''
#     for ch in range(len(string)-1, -1, -1):
#         rev += string[ch]

#     return rev

File: Js_Solutions/reverse_string/str_reverse.js
// Given a string, return a new string
// with its characters reversed

function str_reverse(str) {
    let rev = '';

    for (let i = str.length - 1; i >= 0; i--) {
        rev += str[i];
    }

    return rev;
}


console.log(str_reverse('123456'))

// Alternate solution 1
// function str_reverse(str) {
//     return str.split('').reverse().join('');
// }

// Alternate solution 2
// function str_reverse(str) {
//     let rev = '';

//     for (let character of str) {
//         rev = character + rev;
//     }

//     return rev;
// }
